elide-async/src/test/java/com/yahoo/elide/async/service/AsyncExecutorServiceTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.ElideSettings;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.AsyncQueryResult;
+import com.yahoo.elide.async.models.QueryStatus;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.jsonapi.JsonApiMapper;
+import com.yahoo.elide.security.User;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class AsyncExecutorServiceTest {
+
+    private AsyncExecutorService service;
+    private Elide elide;
+    private AsyncQueryDAO asyncQueryDao;
+
+    @BeforeAll
+    public void setupMocks() {
+        elide = mock(Elide.class);
+        asyncQueryDao = mock(DefaultAsyncQueryDAO.class);
+        ElideSettings elideSettings = mock(ElideSettings.class);
+        EntityDictionary dictionary = mock(EntityDictionary.class);
+        JsonApiMapper mapper = mock(JsonApiMapper.class);
+        ObjectMapper objectMapper = mock(ObjectMapper.class);
+        Set<Class<?>> boundclasses = new HashSet<Class<?>>();
+        boundclasses.add(AsyncQuery.class);
+        boundclasses.add(AsyncQueryResult.class);
+
+        Set<String> apiCollection = new HashSet<String>();
+        apiCollection.add("v1");
+
+        when(elide.getElideSettings()).thenReturn(elideSettings);
+        when(elideSettings.getDictionary()).thenReturn(dictionary);
```

Comment:
Normally, we don't mock the dictionary or the elide settings.  Is there a reason you can't just create a dictionary here and use it?